packages/stablestudio-plugin-comfyui/src/index.ts

```diff
@@ -0,0 +1,615 @@
+    "4": {
+      class_type: "CheckpointLoaderSimple",
+      inputs: {
+        ckpt_name: model,
+      },
+    },
+    "6": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: prompt,
+      },
+    },
+    "7": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: negative_prompt,
+      },
+    },
+    "8": {
+      class_type: "VAEDecode",
+      inputs: {
+        samples: ["3", 0],
+        vae: ["4", 2],
+      },
+    },
+    "9": {
+      class_type: "SaveImage",
+      inputs: {
+        filename_prefix: "ComfyUI",
+        images: ["8", 0],
+      },
+    },
+    "10": {
+      class_type: "LoadImage",
+      inputs: {
+        image: imageFilename,
+        "choose file to upload": "image",
+      },
+    },
+    "11": {
+      class_type: "VAEEncode",
+      inputs: {
+        pixels: ["10", 0],
+        vae: ["4", 2],
+      },
+    },
+  };
+};
+
+const basicWorkflowInpaint = (
+  prompt: string,
+  negative_prompt: string,
+  model: string,
+  seed: number,
+  steps: number,
+  sampler: string,
+  cfgScale: number,
+  imageFilename: string,
+  imageStrength: number,
+  maskFilename: string
+) => {
+  imageStrength = Math.max(0, Math.min(0.99, imageStrength));
+  return {
+    "3": {
+      class_type: "KSampler",
+      inputs: {
+        cfg: cfgScale,
+        denoise: 1 - imageStrength,
+        latent_image: ["17", 0],
+        model: ["4", 0],
+        negative: ["7", 0],
+        positive: ["6", 0],
+        sampler_name: sampler,
+        scheduler: "normal",
+        seed: seed,
+        steps: steps,
+      },
+    },
+    "4": {
+      class_type: "CheckpointLoaderSimple",
+      inputs: {
+        ckpt_name: model,
+      },
+    },
+    "6": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: prompt,
+      },
+    },
+    "7": {
+      class_type: "CLIPTextEncode",
+      inputs: {
+        clip: ["4", 1],
+        text: negative_prompt,
+      },
+    },
+    "8": {
+      class_type: "VAEDecode",
+      inputs: {
+        samples: ["3", 0],
+        vae: ["4", 2],
+      },
+    },
+    "9": {
+      class_type: "SaveImage",
+      inputs: {
+        filename_prefix: "ComfyUI",
+        images: ["8", 0],
+      },
+    },
+    "10": {
+      class_type: "LoadImage",
+      inputs: {
+        image: imageFilename,
+        "choose file to upload": "image",
+      },
+    },
+    "17": {
+      class_type: "VAEEncodeForInpaint",
+      inputs: {
+        grow_mask_by: 0,
+        mask: ["21", 0],
+        pixels: ["10", 0],
+        vae: ["4", 2],
+      },
+    },
+    "20": {
+      class_type: "LoadImageMask",
+      inputs: {
+        channel: "red",
+        "choose file to upload": "image",
+        image: maskFilename,
+      },
+    },
+    "21": {
+      class_type: "InvertMask",
+      inputs: {
+        mask: ["20", 0],
+      },
+    },
+  };
+};
+
+const getExtras = () => {
```

Comment:
Extras code isn't complete, the idea was to allow plugins to define extra UI elements